src/wp-includes/class-wp-block.php

```diff
@@ -139,6 +139,34 @@ public function __construct( $block, $available_context = array(), $registry = n
 		$this->block_type = $registry->get_registered( $this->name );
 
 		$this->available_context = $available_context;
+		$this->update_available_context( $block, $available_context );
```

Comment:
What do you think about making `$available_context` property defaulting to an empty array? In effect, we wouldn't have to set it here and let the value update when merging inside `update_available_context`:
```suggestion
		$this->update_available_context( $block, $available_context );
```

Reply:
If we remove `$this->available_context = $available_context;` then the initialisation for `$this->available_context` is null as it not define https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-block.php#L59 and show error.

```
1) Tests_Blocks_RenderBlock::test_render_block_context_inner_blocks
TypeError: array_merge(): Argument #1 must be of type array, null given

/var/www/src/wp-includes/class-wp-block.php:167
/var/www/src/wp-includes/class-wp-block.php:142
/var/www/src/wp-includes/blocks.php:2188
/var/www/src/wp-includes/blocks.php:2242
/var/www/tests/phpunit/tests/blocks/renderBlock.php:238

2) Tests_Blocks_RenderBlock::test_render_block_context_allowed_context
TypeError: array_merge(): Argument #1 must be of type array, null given

/var/www/src/wp-includes/class-wp-block.php:167
/var/www/src/wp-includes/class-wp-block.php:142
/var/www/src/wp-includes/blocks.php:2188
/var/www/src/wp-includes/blocks.php:2242
/var/www/tests/phpunit/tests/blocks/renderBlock.php:309
```

Reply:
Yes, it requires changes also for the initial value for the `available_context` field so it’s always an array.

Reply:
That would make sense to me, +1 to initialize this property as empty array by default.

Reply:
Update in https://github.com/WordPress/wordpress-develop/pull/7522/commits/33930ed2183e891d752f2019db169e2fd22dde98